src/Analysis/Codelyzer.Analysis.Build/ProjectBuildHandler.cs

```diff
@@ -37,30 +37,45 @@ public class ProjectBuildHandler : IDisposable
 
         private const string syntaxAnalysisError = "Build Errors: Encountered an unknown build issue. Falling back to syntax analysis";
 
-        private List<PortableExecutableReference> LoadMetadataReferences()
+        private XDocument LoadProjectFile(string projectFilePath)
         {
-            XDocument projectFile = null;
-            var references = new List<PortableExecutableReference>();
-            
+            if (!File.Exists(projectFilePath))
+            {
+                return null;
+            }
             try
             {
-                projectFile = XDocument.Load(Project.FilePath);
+                return XDocument.Load(projectFilePath);
             }
             catch (Exception ex)
             {
-                Logger.LogError(ex, "Error loading project file");
-            }   
+                Logger.LogError(ex, "Error loading project file {}", projectFilePath);
+                return null;
+            }
+
+        }
+        private List<PortableExecutableReference> LoadMetadataReferences(XDocument projectFile)
+        {
+            var references = new List<PortableExecutableReference>();
+
+            if (projectFile == null) {
+                return references;
+            }
 
             var fileReferences = ExtractFileReferencesFromProject(projectFile);
             fileReferences?.ForEach(fileRef =>
             {
+                if(!File.Exists(fileRef)) {
+                    Logger.LogWarning("Assembly {} referenced does not exist.", fileRef);
+                    return;
+                }
```

Comment:
Can we add a metric for this if it doesn't exist? This can be part of this release or a new issue you can work on later. We need to keep track of this to make sure we aren't adding wrong references.

Reply:
Metrics has been added [here](https://github.com/aws/cta/pull/199/files). Once this PR gets merged, I'll update the CTA metric PR to include this new version of codelyzer.